### ISSUE TITLE
Symmetrical Simulation Produces an Error

### ISSUE THREAD

**Issue by pjbollinger** [DeviaVir/zenbot#750]
In the current [`master` branch](https://github.com/carlos8f/zenbot/tree/c4fdade3859bf0e1e0df932460969f9f8363e6e2), there is a bug when running a simulation with the option `--symmetrical=true`.

# How to Repeat

1. Open terminal to latest branch and install zenbot.
2. Get some backfill data with `zenbot backfill gdax.ETH-USD --days=1`
3. Run `zenbot sim gdax.ETH-USD --strategy=macd --days=1 --symmetrical=true`

## Error Message

```
/Users/pjbollinger/Downloads/zenbot/node_modules/codemap/index.js:233
        throw err;
        ^

Error: path `zenbot:exchanges.true` is undefined
    at Object.get (/Users/pjbollinger/Downloads/zenbot/node_modules/codemap/index.js:231:19)
    at get (/Users/pjbollinger/Downloads/zenbot/node_modules/codemap/index.js:60:22)
    at /Users/pjbollinger/Downloads/zenbot/lib/engine.js:22:18
    at Command.<anonymous> (/Users/pjbollinger/Downloads/zenbot/commands/sim.js:82:39)
    at Command.listener (/Users/pjbollinger/Downloads/zenbot/node_modules/commander/index.js:300:8)
    at emitTwo (events.js:126:13)
    at Command.emit (events.js:214:7)
    at Command.parseArgs (/Users/pjbollinger/Downloads/zenbot/node_modules/commander/index.js:635:12)
    at Command.parse (/Users/pjbollinger/Downloads/zenbot/node_modules/commander/index.js:457:21)
    at /Users/pjbollinger/Downloads/zenbot/zenbot.js:39:11
    at /Users/pjbollinger/Downloads/zenbot/boot.js:28:7
    at f (/Users/pjbollinger/Downloads/zenbot/node_modules/once/once.js:25:25)
    at Glob.<anonymous> (/Users/pjbollinger/Downloads/zenbot/node_modules/glob/glob.js:151:7)
    at emitOne (events.js:116:13)
    at Glob.emit (events.js:211:7)
    at Glob._finish (/Users/pjbollinger/Downloads/zenbot/node_modules/glob/glob.js:197:8)
    at done (/Users/pjbollinger/Downloads/zenbot/node_modules/glob/glob.js:182:14)
    at Glob._processGlobStar2 (/Users/pjbollinger/Downloads/zenbot/node_modules/glob/glob.js:637:12)
    at /Users/pjbollinger/Downloads/zenbot/node_modules/glob/glob.js:626:10
    at RES (/Users/pjbollinger/Downloads/zenbot/node_modules/inflight/inflight.js:31:16)
    at f (/Users/pjbollinger/Downloads/zenbot/node_modules/once/once.js:25:25)
    at lstatcb_ (/Users/pjbollinger/Downloads/zenbot/node_modules/glob/glob.js:519:7)
```

# Area of where issue starts

I have identified the area of the issue is on [Line 70 of commands/sim.js](https://github.com/carlos8f/zenbot/blob/c4fdade3859bf0e1e0df932460969f9f8363e6e2/commands/sim.js#L70)

``` javascript
so.selector = get('lib.normalize-selector')(selector || c.selector)
```

Based on the results below, this is not the root of the problem, but an area to focus on for resolving it. It appears `--symmetrical=<stuff>` is replacing the `selector` variable in the function call.

# Results of different commands
* `zenbot sim gdax.ETH-USD --strategy=macd --days=1 --symmetrical=true`
   * `selector === 'true'`
   * `c.selector === 'gdax.ETH-USD'`
* `zenbot sim gdax.ETH-USD --strategy=macd --days=1`
   * `selector === 'gdax.ETH-USD'`
   * `c.selector === 'gdax.BTC-USD'`
* `zenbot sim gdax.ETH-USD --strategy=macd --days=1 --symmetrical`
   * `selector === 'gdax.ETH-USD'`
   * `c.selector === 'gdax.BTC-USD'`
* `zenbot sim gdax.ETH-USD --strategy=macd --days=1 --symmetrical=false`
   * `selector === 'false'`
   * `c.selector === 'gdax.BTC-USD'`

**Comment by pjbollinger** [comment]
Doing some more debugging and figuring there is an issue with `commander.js`. Stepping through the Node.js inspector, the first attempt at parsing a command looks good, but the second is bad. I tried debugging without the `--symmetrical=true` and with a different boolean optional, and it worked.

## Bad Output

### Command

```
node-inspect zenbot.js sim gdax.ETH-USD --strategy=macd --days=1 --symmetrical=true
sb('node_modules/commander/index.js',762)
c
exec args
exec unknownOptions
c
exec args
exec unknownOptions
c
.exit
```

### Result

``` javascript
break in node_modules/commander/index.js:762
 760   }
 761 
>762   return { args: args, unknown: unknownOptions };
 763 };
 764 
debug> exec args
[ 'sim', 'gdax.ETH-USD' ]
debug> exec unknownOptions
[ '--strategy',
  'macd',
  '--days',
  '1',
  '--symmetrical',
  'true' ]
debug> c
break in node_modules/commander/index.js:762
 760   }
 761 
>762   return { args: args, unknown: unknownOptions };
 763 };
 764 
debug> exec args
[ 'true' ]
debug> exec unknownOptions
[  ]
```

## Good Output

### Command

```
node-inspect zenbot.js sim gdax.ETH-USD --strategy=macd --days=1
sb('node_modules/commander/index.js',762)
c
exec args
exec unknownOptions
c
exec args
exec unknownOptions
c
.exit
```

### Result

``` javascript
break in node_modules/commander/index.js:762
 760   }
 761 
>762   return { args: args, unknown: unknownOptions };
 763 };
 764 
debug> exec args
[ 'sim', 'gdax.ETH-USD' ]
debug> exec unknownOptions
[ '--strategy',
  'macd',
  '--days',
  '1' ]
debug> c
break in node_modules/commander/index.js:762
 760   }
 761 
>762   return { args: args, unknown: unknownOptions };
 763 };
 764 
debug> exec args
[  ]
debug> exec unknownOptions
[  ]
```

## Good Output with Different Boolean Optional Command

### Command

```
node-inspect zenbot.js sim gdax.ETH-USD --strategy=macd --days=1 --enable_stats
sb('node_modules/commander/index.js',762)
c
exec args
exec unknownOptions
c
exec args
exec unknownOptions
c
Ctrl-C quickly
Ctrl-C quickly
```

### Result

``` javascript
< Debugger attached.
Break on start in zenbot.js:1
> 1 (function (exports, require, module, __filename, __dirname) { 
  2 var semver = require('semver')
  3 var version = require('./package.json').version
debug> sb('node_modules/commander/index.js',762)
Warning: script 'node_modules/commander/index.js' was not loaded yet.
debug> c
break in node_modules/commander/index.js:762
 760   }
 761 
>762   return { args: args, unknown: unknownOptions };
 763 };
 764 
debug> exec args
[ 'sim', 'gdax.ETH-USD' ]
debug> exec unknownOptions
[ '--strategy',
  'macd',
  '--days',
  '1',
  '--enable_stats' ]
debug> c
break in node_modules/commander/index.js:762
 760   }
 761 
>762   return { args: args, unknown: unknownOptions };
 763 };
 764 
debug> exec args
[  ]
debug> exec unknownOptions
[  ]
debug> c
```

**Comment by pjbollinger** [comment]
After much debugging, I discovered an issue with how the `.option()` command was used. When `Boolean` was specified in the arguments list for the function, it caused the variable `c.symmetrical` to default `false`. When I tried to use `--symmetrical=true`, it forced an error (and possible bug) in `commander.js` seen above. After removing `Boolean` from the argument list and using only `--symmetrical` did the option `symmetrical === true` in the option list. The symmetrical code does not work at this moment, but PR #751 should fix the issue of passing the option correctly as a `Boolean`.
